### PR TITLE
Updated the attribute splitting logic to use field_separator

### PR DESCRIPTION
**Pull request by ashiqueps**
## Description

The logic to build the search filter array was hardcoded with splitting the attribute names with a dot only. This was causing issues when there is a dot present in the attribute name, for example a package with version number included. This will not return the correct results. 
```bash
knife search node "packages:libtcl*" -a packages.libtcl8.6.version
```
Eventhough we have an argument `-S(--field-separator SEPARATOR)` to separate the attribute names, because of the hardcoding in the splitting logic, this is also not working. 

I've updated the logic to use the field-separator passed along with the command so that the attributes can be constructed correctly. So, the below command will solve the above issue.

```bash
╰─➤  knife search node "packages:libtcl*" -a packages:libtcl8.6:version -S:
3 items found

node-1:
  packages:libtcl8.6:version: 8.6.12+dfsg-1build1

demo-node:
  packages:libtcl8.6:version: 8.6.12+dfsg-1build1

node-3:
  packages:libtcl8.6:version: 8.6.12+dfsg-1build1
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by nikhil2611 — APPROVED** [review]

**Comment by tpowell-progress** [comment]
@ashiqueps Can you rebase on current `main` and also add a test for the change to cover the default and field separator specified case?

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Please rebase on main and add test for default and alternative separator case.

**Comment by ashiqueps** [comment]
Hi @tpowell-progress,

I've rebased with main and added specs for the field_separator. Please review.

**Review by sanjain-progress — APPROVED** [review]

File: knife/spec/unit/knife/core/ui_spec.rb [hunk 1]
```diff
@@ -430,6 +430,15 @@ def self.from_hash(hsh)
           expect(@ui.format_for_display(input)).to eq({ nil => { "keys;with spaces;open;doors;with many.dots" => "when asked" } })
         end
       end
+
+      it "should return field separator in attribute_field_separator" do
```
Comment (tpowell-progress) [inline knife/spec/unit/knife/core/ui_spec.rb:434]: I think I'd rather see that `create_result_filter_from_attributes` responds to a set/unset `attribute_field_separator` vs. testing the set/get of the field separator itself.

**Review by tpowell-progress — APPROVED** [review]

**Pull request merged**